src/core/ghosts.cpp

```diff
@@ -107,87 +161,60 @@ int calc_transmit_size(GhostCommunication *gc, int data_parts) {
     if (data_parts & GHOSTTRANS_SWIMMING)
       n_buffer_new += sizeof(ParticleParametersSwimming);
 #endif
+    int count = 0;
+    for (auto const &pl : gc.part_lists)
+      count += pl->n;
     n_buffer_new *= count;
   }
-  // also sending length of bond buffer
-  if (data_parts & GHOSTTRANS_PROPRTS)
-    n_buffer_new += sizeof(int);
   return n_buffer_new;
 }
 
-void prepare_send_buffer(GhostCommunication *gc, int data_parts) {
+static void prepare_send_buffer(CommBuf &s_buffer, GhostCommunication &gc,
+                                int data_parts) {
   /* reallocate send buffer */
-  n_s_buffer = calc_transmit_size(gc, data_parts);
-  if (n_s_buffer > max_s_buffer) {
-    max_s_buffer = n_s_buffer;
-    s_buffer = Utils::realloc(s_buffer, max_s_buffer);
-  }
+  s_buffer.resize(calc_transmit_size(gc, data_parts));
```

Comment:
a lot of the code would be easier to read if the variable names would be just a little more descriptive: *e.g.*:
 * ```send_buffer``` instead of ```s_buffer```,
 * ```rec_buffer``` instead of ```r_buffer```,
 * ```ghost_comm``` instead of ```gc```,
 * ```archiver``` instead of ```ar```,
 * ```bond_archiver``` instead of ```bar```,
 * ```bond_list``` instead of ```bl```,
 * ```part``` instead of ```pt```,
 * ```part_list``` instead of ```cur_list```